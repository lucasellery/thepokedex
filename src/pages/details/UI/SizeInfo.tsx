import { observer } from 'mobx-react';
import React from 'react';
import { StyleSheet } from 'react-native';
import { View, Text } from 'react-native';

// import { Container } from './styles';

export type SizeInfoProps = {
  weight: number;
  height: number;
  baseExperience: number;
}

const SizeInfo: React.FC<SizeInfoProps> = observer(({ weight, height, baseExperience }) => {
  return (
    <View style={styles.container} testID="size-info">
      <View>
        <Text style={styles.information} testID="height-info">{height ? `${height / 10}m`.replace('.', ',') : '?'}</Text>
        <Text>Height</Text>
      </View>
      <View>
        <Text style={styles.information} testID="weight">{weight ? `${weight / 10}kg` : '?'}</Text>
        <Text>Weight</Text>
      </View>
      <View>
        <Text style={styles.information} testID="base-experience-info">{baseExperience ? baseExperience : '?'}</Text>
        <Text>Base experience</Text>
      </View>
    </View>
  );
})

export default SizeInfo;

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingTop: 20
  },
  information: {
    fontWeight: 'bold',
    fontSize: 20,
    lineHeight: 26,
    textAlign: 'center'
  },
  label: {
    fontWeight: 'normal',
    fontSize: 12,
    lineHeight: 16,
    textAlign: 'center',
  }
})
