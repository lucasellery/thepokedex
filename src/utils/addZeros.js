function addZeros(number) {
  const amountOfZeros = {
    [3]: `${number}`,
    [2]: `0${number}`,
    [1]: `00${number}`,
  }

  return amountOfZeros[number?.length] || `${0}`
}

export default addZeros;