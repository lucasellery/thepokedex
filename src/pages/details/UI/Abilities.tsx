import React from "react";
import { useEffect } from "react";
import { View, Text, StyleSheet } from "react-native";
import { toCaptalize } from "../../../utils/toCaptalize";

export type Ability = {
  ability: {
    name: string;
  };
};

export type AbilityProps = {
  abilities: Ability[];
};

const Abilities: React.FC<AbilityProps> = ({ abilities }) => {
  useEffect(() => {
    abilities?.map(({ ability }) => {
      console.log(toCaptalize(ability?.name))
    });
  }, [])

  return (
    <View style={styles.container} testID="abilities-component">
      <Text style={styles.title}>Abilities</Text>

      <View style={styles.wrapper}>
        {abilities?.map(({ ability }) => (
          <Text style={styles.ability} key={ability?.name} testID="ability">
            {toCaptalize(ability?.name)}
          </Text>
        ))}
      </View>
    </View>
  );
};

export default Abilities;

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
  wrapper: {
    marginTop: 10,
    flexDirection: "row",
    justifyContent: "center",
  },
  title: {
    fontWeight: "bold",
    textAlign: "center",
    fontSize: 16,
  },
  ability: {
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: "#F7F7F7",
    borderRadius: 20,
    overflow: "hidden",
    alignItems: "center",
    marginHorizontal: 5,
  },
});
