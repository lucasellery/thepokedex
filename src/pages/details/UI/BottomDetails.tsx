import React, { useEffect } from 'react'
import { StyleSheet, View } from 'react-native';
import SizeInfo, { SizeInfoProps } from './SizeInfo';
import { usePkmStore } from '../../../mobx/pkmProvider';
import { observer } from 'mobx-react';
import Abilities, { Ability } from './Abilities';
import Stats from './Stats';

type BottomDetailsProps = {
  sizeInfo: SizeInfoProps
  abilities: Ability[]
  pokemonTypeColor: {
    main: string
  }
  stats: Array<{
    base_stat: number,
    stat: {
      name: string
    }
  }>
}

const BottomDetails: React.FC<BottomDetailsProps> = ({ sizeInfo, abilities, stats, pokemonTypeColor }) => {
  return (
    <View testID="bottom-details" style={styles.wrapper}>
      <SizeInfo weight={sizeInfo?.weight} height={sizeInfo?.height} baseExperience={sizeInfo?.baseExperience} />
      <Abilities abilities={abilities} />
      <Stats pokemonTypeColor={pokemonTypeColor} stats={stats} />
    </View>
  );
}

export default BottomDetails;

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: 'white',
    borderTopLeftRadius: 30,
    borderTopRightRadius:30,
    marginTop: -30
  }
})
