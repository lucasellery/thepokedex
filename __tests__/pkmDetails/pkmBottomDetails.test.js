import React from 'react'
import { render } from '@testing-library/react-native'

import BottomDetails from '../../src/pages/details/UI/BottomDetails';
import SizeInfo from '../../src/pages/details/UI/SizeInfo';
import Abilities from '../../src/pages/details/UI/Abilities';
import { toCaptalize } from '../../src/utils/toCaptalize';

describe("Pkm Details Bottom Tests", () => {
    
	it('Should render bottom correcly', () => {
		const { queryByTestId } = render(<BottomDetails />)
		const component = queryByTestId('bottom-details')
		expect(component).toBeTruthy()
	})

	it('Should render SizeInfo correctly', () => {
		const { queryByTestId } = render(<SizeInfo />)
		const component = queryByTestId('size-info')
		expect(component).toBeTruthy()
	})

	it('Should render SizeInfo weight correcly', () => {
		const weight = 60

		const { queryByTestId } = render(<SizeInfo weight={weight} />)
		const component = queryByTestId('weight')
		expect(component.props.children).toBe('6kg')
	})

	it('should render SizeInfo height correctly', async ()=> {
		const height = 70

		const { queryByTestId } = render(<SizeInfo height={height} />)
		const heightProp = queryByTestId('height-info')
		expect(heightProp.props.children).toBe('7m')
	})

	it('should render SizeInfo height correctly', async () => {
		const baseExperience = 112

		const { queryByTestId } = render(<SizeInfo baseExperience={baseExperience} />)
		const baseXpProp = queryByTestId('base-experience-info')
		expect(baseXpProp.props.children).toBe(baseExperience)
	})

	it('should match snapshot <SizeInfo />', () => {
    const weight = 60
    const height = 70
    const baseExpericence = 112
    const component = render(
      <SizeInfo
        baseExpericence={baseExpericence}
        height={height}
        weight={weight} />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

	// Abilities

	it('should render Abilities correctly', async () => {
		const { queryByTestId } = render(<Abilities />)
		const component = queryByTestId('abilities-component')
		expect(component).toBeTruthy()
	})

	it('should render an array of abilities', () => {
		const abilities = [
			{
				ability: {
					name: 'static'
				}
			},
			{
				ability: {
					name: 'poison pin'
				}
			},
		]

		const { queryAllByTestId } = render(<Abilities abilities={abilities} />)
		const componentArray = queryAllByTestId('ability')
		expect(componentArray.length).toBeGreaterThan(0)

		componentArray.forEach((comp, index) => {
			expect(comp.props.children).toBe(toCaptalize(abilities[index].ability.name))
		})

	})


})