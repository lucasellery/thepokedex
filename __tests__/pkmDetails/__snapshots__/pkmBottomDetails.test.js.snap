// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pkm Details Bottom Tests should match snapshot <SizeInfo /> 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexGrow": 1,
      "justifyContent": "space-around",
      "paddingTop": 20,
    }
  }
  testID="size-info"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="height-info"
    >
      7m
    </Text>
    <Text>
      Height
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="weight"
    >
      6kg
    </Text>
    <Text>
      Weight
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="base-experience-info"
    >
      ?
    </Text>
    <Text>
      Base XP
    </Text>
  </View>
</View>
`;
